### PR TITLE
chore(root): Framework v2.0.3

### PR DESCRIPTION
**Pull request by SokratisVidros**
### What changed? Why was the change needed?
Latest framework release plus a script to help us restore the workspace:* protocol

### CODE REVIEW

File: scripts/restore-workspace-protocol.mjs [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import fs from 'fs-extra';
+import path from 'node:path';
+import { glob } from 'glob';
+import {execa} from 'execa';
```
Comment (rifont) [inline scripts/restore-workspace-protocol.mjs]: Small whitespace fix

```suggestion
import { execa } from 'execa';
```

File: scripts/restore-workspace-protocol.mjs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+  updateNovuDependencies(packageJson.dependencies);
+  updateNovuDependencies(packageJson.devDependencies);
+
+  fs.writeJsonSync(filePath, packageJson, { spaces: 2 });
+  console.log(`Restored workspace protocol at ${filePath}`);
+}
+
+// Find all package.json files in the repo
+const files = await glob('packages/**/package.json', { ignore: 'packages/**/node_modules/**' });
```
Comment (rifont) [inline scripts/restore-workspace-protocol.mjs:37]: Is it intentional to target `packages` directory only?

File: scripts/restore-workspace-protocol.mjs [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      dependencies[key] = 'workspace:*';
+    }
+  }
+}
+
+function processPackageJson(filePath) {
+  const packageJson = fs.readJsonSync(filePath);
+
+  updateNovuDependencies(packageJson.dependencies);
+  updateNovuDependencies(packageJson.devDependencies);
```
Comment (rifont) [inline scripts/restore-workspace-protocol.mjs:28]: we need to cover `peerDependencies` and `optionalDependencies` too, as some packages use them ([ref](https://github.com/novuhq/novu/blob/next/enterprise/packages/billing/package.json#L52))

File: scripts/restore-workspace-protocol.mjs [hunk 4]
```diff
@@ -0,0 +1,38 @@
+import fs from 'fs-extra';
+import path from 'node:path';
+import { glob } from 'glob';
+import {execa} from 'execa';
+
+// The following node packages don't belong to the monorepo and must always be references with specific semver.
+const IGNORE_LIST = ["@novu/ntfr-client", "@novu/floating-vue"];
```
Comment (antonjoel82) [inline scripts/restore-workspace-protocol.mjs]:
```suggestion
const IGNORE_LIST = new Set(["@novu/ntfr-client", "@novu/floating-vue"]);
```

File: scripts/restore-workspace-protocol.mjs [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+async function getPackageNames() {
+  const { stdout } = await execa`pnpm nx show projects`;
+  return stdout;
+}
+
+
+// Function to update versions of @novu dependencies
+function updateNovuDependencies(dependencies) {
+  for (const [key] of Object.entries(dependencies || {})) {
+    if (key.startsWith('@novu/') && IGNORE_LIST.indexOf(key) === -1) {
```
Comment (antonjoel82) [inline scripts/restore-workspace-protocol.mjs]:
```suggestion
    if (key.startsWith('@novu/') && IGNORE_LIST.has(key)) {
```

File: scripts/restore-workspace-protocol.mjs (hunk 2 above)
Comment (SokratisVidros) [reply scripts/restore-workspace-protocol.mjs:37]: For now yes, as they are the only ones that we change the workspace during publishing.

File: scripts/restore-workspace-protocol.mjs (hunk 1 above)
Comment (SokratisVidros) [reply scripts/restore-workspace-protocol.mjs]: Fixed

File: scripts/restore-workspace-protocol.mjs (hunk 3 above)
Comment (SokratisVidros) [reply scripts/restore-workspace-protocol.mjs:28]: Good call.

File: scripts/restore-workspace-protocol.mjs (hunk 4 above)
Comment (SokratisVidros) [reply scripts/restore-workspace-protocol.mjs]: Fixed

**Review by LetItRock — APPROVED** [review]

**Pull request merged**